com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -285,7 +282,16 @@ protected String getIndexVariable() {
 
     @Override
     public Integer getTimeout(DelegateExecution context) {
-        return configuration.getUploadAppTimeout();
+        CloudApplication app = StepsUtil.getApp(context);
+        int uploadTimeout = extractUploadTimeoutFromAppAttributes(app, DEFAULT_APP_UPLOAD_TIMEOUT);
+        getStepLogger().debug(Messages.UPLOAD_APP_TIMEOUT, uploadTimeout);
+        return uploadTimeout;
+    }
+
+    private int extractUploadTimeoutFromAppAttributes(CloudApplication app, int defaultAppUploadTimeout) {
+        ApplicationAttributes appAttributes = ApplicationAttributes.fromApplication(app);
+        Number uploadTimeout = appAttributes.get(SupportedParameters.UPLOAD_TIMEOUT, Number.class, defaultAppUploadTimeout);

```

Comment:
I am wondering whether this is consistent with the CF push approach? There is no upload-timeout but instead there is just timeout or start timeout which are applied either for just one operation or the whole push operation?

From the user POV, it will be easier to configure the timeout for each application, but on the other hand, is it necessary?

Why is bad to leave it to be configured from the ENV of deploy-service?

Reply:
It is not necessarily bad, it's just that this is a better and more flexible approach. It allows users to specify smaller timeouts for applications they know are small in size, while also allowing them to put bigger timeouts on large applications (SAPUI5, for example).

cf push does not have an upload timeout at all, but since we do, it's better if the user can configure it without involving us.